### DISCUSSION TITLE
CookieCutter Django Restore Script not allowed with DigitalOcean Managed Database

### DISCUSSION THREAD

**Discussion by BenA-SA** [cookiecutter/cookiecutter-django#4932] [Q&A] [answered]
Hi all,

I'll start this by saying I have exclusively used Django's ORM, and have little knowledge of Postgres itself, so apologies if this is a stupid/simple question.

I have a Cookiecutter Django project I made ~2 years ago (with Docker, if that's relevant) for my job at work. The database is getting really big now, and I wanted to migrate it to a DigitalOcean Managed Database.

I used the backup script to get my backup file, but I am unable to use the restore script to add the data to my new managed database. DigitalOcean say this is because the restore script is trying to drop the whole database, which they do not allow for as they need certain parts of the database to remain for their functionality.

Is there a way I can adjust the restore script to simply insert the data without dropping the database, or are there reasons that this wouldn't work?

Thanks for any help anyone can offer!

**Comment by foarsitter** [comment, marked as answer]
The `.sql.gz` file is a gzip version of the `pg_dump` output. If you `gunzip` it you can use it with your favorite database tools. 

Another option is to just remove the `dropdb` and `createdb` command from https://github.com/cookiecutter/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/compose/production/postgres/maintenance/restore

**Reply by BenA-SA** [reply]
Removing the dropdb and createdb commands has done the trick. Thanks so much for your help.

Are there any potential issues that may arise from removing those commands, or should the restore function still work properly?

I got some errors when I ran the command (multiple primary keys, some relations already existing), but I think this is because I had already run migrate on the new database? It seems like all the data has been imported properly, but I'll do more testing to check.

Thanks again!

**Reply by foarsitter** [reply]
If you want to be sure delete the digitalocean managed database instance and create a new one, then you are sure the restore is correct.

**Reply by BenA-SA** [reply]
So I would create the managed database, not run the migrate command, and run the restore command? Apologies for the basic question, I just want to be crystal clear on what I should do of course. Thanks for your help!

**Reply by luzfcb** [reply]
> So I would create the managed database, not run the migrate command, and run the restore command

Yes.
